clang/lib/Driver/ToolChains/Arch/RISCV.cpp

```diff
@@ -86,8 +86,14 @@ void riscv::getRISCVTargetFeatures(const Driver &D, const llvm::Triple &Triple,
   // and other features (ex. mirco architecture feature) from mcpu
   if (Arg *A = Args.getLastArg(options::OPT_mcpu_EQ)) {
     StringRef CPU = A->getValue();
-    if (CPU == "native")
+    if (CPU == "native") {
       CPU = llvm::sys::getHostCPUName();
+      llvm::StringMap<bool> HostFeatures;
+      if (llvm::sys::getHostCPUFeatures(HostFeatures))
```

Comment:
Open discussion here: `CPU` may fail and return `generic`. Should we failback to use `getHostCPUFeatures` if `getHostCPUName` fails? Or we should use `getHostCPUFeatures` all the time?

Reply:
I think hwprobe is more reliable than cpu name because some RV cpus are not recognized by LLVM (e.g., T-head's cores). However, cpu name is still useful even if `getHostCPUFeatures` returns true because it may provide scheduling model.